From fe56124164ef5f4f39f499287f487464a6113b48 Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Mon, 12 Aug 2019 12:38:30 +0200
Subject: [PATCH] arm64: dt: fsl-imx8mq-tqma8mq: add regulator for CPU
 overdrive mode

Model it with gpio-regulator as done in imx8mq-evk.

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 .../dts/freescale/fsl-imx8mq-tqma8mq.dtsi     | 34 ++++++++++++++-----
 1 file changed, 26 insertions(+), 8 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq.dtsi
index 733ec6c903e6..a8b12351e332 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq.dtsi
@@ -26,14 +26,6 @@
 	chosen {
 	};
 
-	reg_tqma8mx_vcc3v3: tqma8mx_vcc3v3 {
-		compatible = "regulator-fixed";
-		regulator-name = "TQMA8MX_VCC3V3";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
-		regulator-always-on;
-	};
-
 	memory@40000000 {
 		device_type = "memory";
 		/*  our minimum RAM config will be 512 MiB */
@@ -60,6 +52,26 @@
 		/delete-node/ rpmsg@0x78000000;
 	};
 
+	reg_tqma8mx_vcc3v3: tqma8mx-vcc3v3 {
+		compatible = "regulator-fixed";
+		regulator-name = "TQMA8MX_VCC3V3";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		regulator-always-on;
+	};
+
+	reg_tqma8mx_dvfs: tqma8mx-vdd-arm {
+		compatible = "regulator-gpio";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_dvfs>;
+		regulator-min-microvolt = <900000>;
+		regulator-max-microvolt = <1000000>;
+		regulator-name = "TQMa8Mx DVFS";
+		regulator-type = "voltage";
+		gpios = <&gpio1 6 GPIO_ACTIVE_HIGH>;
+		states = <900000 0x1 1000000 0x0>;
+	};
+
 	thermal-zones {
 		cpu-thermal {
 			thermal-sensors = <&tmu 0>;
@@ -79,6 +91,12 @@
 	pinctrl-names = "default";
 
 	imx8mq-tqma8mq {
+		pinctrl_dvfs: dvfsgrp {
+			fsl,pins = <
+				MX8MQ_IOMUXC_GPIO1_IO06_GPIO1_IO6	0x16
+			>;
+		};
+
 		pinctrl_i2c1: i2c1grp {
 			fsl,pins = <
 				MX8MQ_IOMUXC_I2C1_SCL_I2C1_SCL			0x4000007f
